File: kitsune/test/driver/empty.m
// Kitsune does not support Objective C

// RUN: %clang -c -o /dev/null %s
// RUN: %kitxx -c -o /dev/null %s
// RUN: not %kitxx -ftapir=serial -O1 %s 2>&1 | FileCheck %s

// CHECK: kitsune does not support Objective-C

File: kitsune/test/forall/empty.c
// RUN: %clang -ftapir=none -S -emit-llvm -o - %s | FileCheck %s

#include <kitsune.h>

void loop(int* a) {
  // Empty forall's are ok.
  forall(int i = 0; i < 10; i++) {
  }
}

// CHECK: %[[SYNCREG:.+]] = call token @llvm.syncregion.start()
// CHECK: detach within %[[SYNCREG:.+]], label %[[BODY:.+]], label %[[INC:.+]]
// CHECK: [[BODY]]
// CHECK store
// CHECK: br label %[[REATTACH:.+]]
// CHECK: [[REATTACH]]:
// CHECK: reattach within %[[SYNCREG]], label %[[INC]]
// CHECK: [[INC]]:
// CHECK: br label {{.+}}, !llvm.loop
// CHECK: [[SYNC:.+]]:
// CHECK: sync within %[[SYNCREG]]
